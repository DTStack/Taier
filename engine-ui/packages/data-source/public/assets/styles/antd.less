@import '~antd/dist/antd.less';

@primary-color: #3f87ff; // 全局主色
@success-color: #12bc6a; // 成功色
@warning-color: #ffb310; // 警告色
@error-color: #ff5f5c; // 错误色
@font-size-base: 12px; // 主字号
@link-color: #5c99ff; // 链接色
@heading-color: #333; // 标题色
@text-color: #333; // 主文本色
@text-color-secondary: #666; // 次文本色
@disabled-color: #bfbfbf; // 失效色
@border-color-base: #ddd; // 边框色
// @box-shadow-base: 0 2px 8px rgba(0, 0, 0, 0.15); // 浮层阴影

// antd 分页器样式
.ant-pagination-item-active {
	a {
		font-family: PingFangSC-Regular;
		letter-spacing: 0;
		text-align: center;
	}
}

.ant-pagination {
	position: relative;
	left: 9px;
	box-sizing: border-box;
	margin: 0;
	margin: 14px 10px 0 0;
	padding: 0;
	color: rgba(0, 0, 0, 0.85);
	font-size: 14px;
	font-variant: tabular-nums;
	line-height: 1.5715;
	list-style: none;
	font-feature-settings: 'tnum';
}

.ant-table-cell {
	border-right: none !important;
}
.ant-table-measure-row td > div {
	display: none !important;
}
.dt-notification-close-btn {
	position: absolute !important;
	top: 18px !important;
	right: 46px;
	font-weight: normal;
	font-size: 13px;
	line-height: 17px;
	user-select: none;
	&:hover {
		color: rgb(39, 39, 39);
	}
}
.anticon-eye-invisible .ant-input-password-icon {
	color: #999;
}
.anticon-close .ant-select-remove-icon {
	color: #999;
}

// 输入框，下拉框禁用样式
.ant-input[disabled] {
	color: #999999 !important;
	background-color: #e8e8e8 !important;
	border: 1px solid #dddddd;
}

.ant-select-disabled {
	&:hover {
		border: none;
	}
	.ant-select-selection {
		color: #999999 !important;
		background-color: #e8e8e8 !important;
		border: 1px solid #dddddd;
	}
}

.ant-form-item {
	margin-bottom: 20px;

	&.ant-form-item-with-help {
		margin-bottom: 4px;
	}

	label {
		top: 50%;
		display: inline-block;
		transform: translateY(-50%);
	}

	.ant-form-item-label {
		height: 32px;
	}

	.ant-form-item-label,
	.ant-form-item-control {
		line-height: unset;
	}

	textarea.ant-input {
		margin-bottom: 0;
	}
}

.ant-modal {
	height: 80vh;

	.ant-modal-content {
		max-height: 80vh;
	}

	.ant-modal-body {
		max-height: calc(80vh - 127px);
		padding-bottom: 28px;
		overflow-y: auto;
	}

	.ant-modal-footer,
	.ant-modal-body,
	.ant-modal-header {
		padding-right: 20px;
		padding-left: 20px;
		border-color: #ddd;
	}

	.ant-modal-footer {
		padding: 19.5px 16px;
	}
}

.ant-table-tbody > tr > td {
	vertical-align: middle;
}

.ant-btn:hover,
.ant-btn:focus,
.ant-btn:active,
.ant-btn.active {
	color: #333;
	border-color: #bfbfbf;
}
.ant-btn.ant-btn-primary:hover,
.ant-btn.ant-btn-primary:focus {
	color: #fff;
	background-color: #5c99ff;
	border-color: #5c99ff;
}
.ant-btn.ant-btn-background-ghost:hover,
.ant-btn.ant-btn-background-ghost:focus {
	color: #fff;
	background-color: #5c99ff !important;
	border-color: #5c99ff;
}

.ant-message-notice-content {
	height: auto !important;
	.ant-message-custom-content {
		height: auto !important;
	}
}
