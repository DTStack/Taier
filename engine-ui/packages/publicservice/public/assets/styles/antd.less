@import '~antd/dist/antd.less';
@main-color: #268BFC;

// antd 分页器样式
.ant-pagination-item {
  &:hover {
    border-color: @main-color  !important;

    a {
      color: @main-color  !important;
    }
  }
}

.ant-pagination-item-active {
  background: @main-color  !important;

  a {
    font-family: PingFangSC-Regular;
    font-size: 14px;
    color: #ffffff !important;
    letter-spacing: 0;
    text-align: center;
  }

  &:hover {
    border-color: @main-color  !important;

    a {
      color: #ffffff !important;
    }
  }
}

.ant-pagination {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  position: relative;
  left: 9px;
  color: rgba(0, 0, 0, 0.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-feature-settings: 'tnum';
  // margin: 10px;
  margin: 14px 10px 0 0;
}

.ant-table-cell {

  border-right: none !important;

}
.ant-table-measure-row td>div{
	display: none !important;
}