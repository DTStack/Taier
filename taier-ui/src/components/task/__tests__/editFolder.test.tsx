import { fireEvent, render, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import { Input } from 'antd';
import EditFolder from '../editFolder';

jest.mock('../../../components/folderPicker', () => {
    return (props: any) => <Input data-testid="folderPicker" {...props} />;
});

describe('Test EditFolder Component', () => {
    it('Should match snapshot', () => {
        const { asFragment } = render(<EditFolder tabId={1} />);

        expect(asFragment()).toMatchSnapshot();
    });

    describe('Test EditFolder Without record', () => {
        it('Should submit successfully', async () => {
            const fn = jest.fn().mockResolvedValue(true);
            const { container, getByTestId } = render(<EditFolder tabId={1} onSubmitFolder={fn} />);

            fireEvent.change(container.querySelector('input#dt_nodeName')!, {
                target: { value: 'test' },
            });

            fireEvent.change(getByTestId('folderPicker'), {
                target: { value: '1' },
            });

            act(() => {
                // submit
                fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);
            });

            await waitFor(() => {
                expect(fn).toBeCalledWith({
                    nodeName: 'test',
                    nodePid: '1',
                });
            });
        });

        it('Should log error on fields', async () => {
            const { container, getByText } = render(<EditFolder tabId={1} />);

            // submit
            fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);

            await waitFor(() => {
                expect(getByText('文件夹名称不能为空')).toBeInTheDocument();
                expect(getByText("'nodePid' is required")).toBeInTheDocument();
            });

            fireEvent.change(container.querySelector('input#dt_nodeName')!, {
                target: { value: new Array(100).fill('a').join('') },
            });

            act(() => {
                // submit
                fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);
            });

            await waitFor(() => {
                expect(getByText('目录名称不得超过64个字符！')).toBeInTheDocument();
            });
        });
    });

    describe('Test EditFolder With record', () => {
        it('Should set fields value with record', async () => {
            const { container, getByTestId } = render(
                <EditFolder tabId={1} current={{ tab: { data: { id: 1, dt_nodeName: 'test', nodePid: '1' } } }} />
            );

            await waitFor(() => {
                expect(container.querySelector<HTMLInputElement>('input#dt_nodeName')?.value).toBe('test');
                expect((getByTestId('folderPicker') as HTMLInputElement).value).toBe('1');
            });
        });
    });
});
