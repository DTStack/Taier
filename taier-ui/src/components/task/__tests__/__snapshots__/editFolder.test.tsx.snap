// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EditFolder Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="Scrollbar"
  >
    <form
      class="ant-form ant-form-horizontal mo-open-task"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
          >
            <label
              class="ant-form-item-required"
              for="dt_nodeName"
              title="目录名称"
            >
              目录名称
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  autocomplete="off"
                  class="ant-input"
                  id="dt_nodeName"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
          >
            <label
              class="ant-form-item-required"
              for="nodePid"
              title="选择目录位置"
            >
              选择目录位置
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  data-testid="folderPicker"
                  datatype="task"
                  id="nodePid"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-14 ant-col-sm-offset-8"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    确 认
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
